src/main/java/de/muenchen/dave/services/messstelle/MessstelleReceiver.java

```diff
@@ -69,16 +69,30 @@ private void processingMessstellenCron(final List<MessstelleDto> messstellen) {
 
     private void createMessstelleCron(final MessstelleDto dto) {
         log.info("#createMessstelleCron");
-        final Messstelle newMessstelle = messstelleReceiverMapper.dtoToMessstelle(dto);
-        newMessstelle.setId(UUID.randomUUID().toString());
-        newMessstelle.getMessquerschnitte().forEach(messquerschnitt -> messquerschnitt.setId(UUID.randomUUID().toString()));
+        final Messstelle newMessstelle = messstelleReceiverMapper.createMessstelle(dto);
         customSuggestIndexService.createSuggestionsForMessstelle(newMessstelle);
         messstelleIndexService.saveMessstelle(newMessstelle);
     }
 
     private void updateMessstelleCron(final Messstelle existingMessstelle, final MessstelleDto dto) {
         log.info("#updateMessstelleCron");
         final Messstelle updated = messstelleReceiverMapper.updateMessstelle(existingMessstelle, dto);
+        updated.getMessquerschnitte().forEach(messquerschnitt -> {
+            if (CollectionUtils.isNotEmpty(dto.getMessquerschnitte())) {
+                dto.getMessquerschnitte().forEach(dto1 -> {
+                    boolean doesNotExist = true;
+                    if (messquerschnitt.getMqId().equalsIgnoreCase(dto1.getMqId())) {
+                        messstelleReceiverMapper.updateMessquerschnitt(messquerschnitt, dto1);
+                        doesNotExist = false;
+                    }
+                    if (doesNotExist) {
+                        updated.getMessquerschnitte().add(messstelleReceiverMapper.createMessquerschnitte(dto1));
+                    }
```

Comment:
kannst du dir hier das zweite `else` und `doesNotExist` sparen und ein if-else drauß machen?

Reply:
Ne da ist ein Fehler in der Logik. Muss ich umbauen